pootle/apps/pootle_store/views.py

```diff
@@ -437,6 +333,181 @@ def source_language(self):
+                         SubmissionFields.COMMENT, SubmissionFields.NONE])
+            | Q(type__in=SubmissionTypes.SUGGESTION_TYPES))
+        return (
+            Submission.objects.filter(unit=self.object)
+                              .filter(submission_filter)
+                              .exclude(field=SubmissionFields.COMMENT,
+                                       creation_time=self.object.commented_on)
+                              .order_by("id"))
+
+    @property
+    def language(self):
+        return self.object.store.translation_project.language
+
+    def add_creation_entry(self, grouped_entries):
+        User = get_user_model()
+        has_creation_entry = (
+            len(grouped_entries) > 0
+            and grouped_entries[0]['datetime'] == self.object.creation_time)
+        if (has_creation_entry):
+            grouped_entries[0]['created'] = True
+        else:
+            created = {
+                'created': True,
+                'submitter': User.objects.get_system_user()}
+            if self.object.creation_time:
+                created['datetime'] = self.object.creation_time
+            grouped_entries[:0] = [created]
+        return grouped_entries
+
+    def get_context_data(self, *args, **kwargs):
+        grouped_entries = self.get_grouped_timeline_entries()
+        grouped_entries = self.add_creation_entry(grouped_entries)
+        grouped_entries.reverse()
+        return dict(
+            entries_group=grouped_entries,
+            language=self.language)
+
+    def render_timeline(self, context):
+        return loader.get_template(
+            self.template_name).render(context).replace('\n', '')
+
+    def get_response_data(self, context):
+        return {
+            'uid': self.object.id,
+            'timeline': self.render_timeline(context)}
+
+    def get_timeline_entry(self, item):
+        entry = {
+            'field': item['field'],
+            'field_name': SubmissionFields.NAMES_MAP.get(item['field'], None),
+            'type': item['type']}
+        if item['field'] == SubmissionFields.STATE:
+            entry['old_value'] = STATES_MAP[int(to_python(item['old_value']))]
+            entry['new_value'] = STATES_MAP[int(to_python(item['new_value']))]
+        elif item['suggestion_id']:
+            entry.update(
+                {'suggestion_text': item['suggestion__target_f'],
+                 'suggestion_description':
+                     mark_safe(self._get_suggestion_description(item))})
+        elif item['quality_check__name']:
+            check_name = item['quality_check__name']
+            check_url = (
+                u''.join(
+                    [reverse('pootle-checks-descriptions'),
+                     '#', check_name]))
+            entry.update(
+                {'check_name': check_name,
+                 'check_display_name': check_names[check_name],
+                 'checks_url': check_url})
```

Comment:
To reiterate what has been said before, this is about keeping consistency throughout the codebase and sticking to a common practice used throughout years in all the major projects under the translate umbrella (TTK, virtaal, pootle). It's true that the previous lack of linting unfortunately contributed to nitpicky reviews and no one wants or is suggesting to go back there. Nevertheless here we're facing a major inconsistency in multi-line dicts, lists and calls, which is a recent thing.

Needless to say it's in the project's best interest and it's everyone's responsibility to keep the code consistently readable, and not contribute to a mess which becomes uncomfortable.